python/pyspark/sql/catalog.py

```diff
@@ -288,19 +301,75 @@ def functionExists(self, functionName: str, dbName: Optional[str] = None) -> boo
             name of the database to check function existence in.
             If no database is specified, the current database is used
 
+           .. deprecated:: 3.4.0
```

Comment:
@amaliujia we need to do the same deprecation in scala

Reply:
ack will follow up on the scala side.